boa/src/context.rs

```diff
@@ -622,25 +636,46 @@ impl Context {
     ///
     /// # Example
     /// ```
-    /// use boa::{Context, property::Attribute, object::ObjectInitializer};
+    /// use boa::{Context, property::{Attribute, PropertyDescriptor}, object::ObjectInitializer};
     ///
     /// let mut context = Context::new();
     ///
-    /// context.register_global_property("myPrimitiveProperty", 10, Attribute::all());
+    /// context.register_global_property(
+    ///     "myPrimitiveProperty",
+    ///     PropertyDescriptor::builder()
+    ///         .value(10)
+    ///         .writable(true)
+    ///         .enumerable(true)
+    ///         .configurable(true)
+    /// );
     ///
     /// let object = ObjectInitializer::new(&mut context)
-    ///    .property("x", 0, Attribute::all())
-    ///    .property("y", 1, Attribute::all())
+    ///    .property(
+    ///         "x",
+    ///         0,
+    ///         Attribute::all()
+    ///     )
+    ///     .property(
+    ///         "y",
+    ///         1,
+    ///         Attribute::all()
+    ///     )
     ///    .build();
-    /// context.register_global_property("myObjectProperty", object, Attribute::all());
+    /// context.register_global_property(
+    ///     "myObjectProperty",
+    ///     PropertyDescriptor::builder()
+    ///         .value(object)
+    ///         .writable(true)
+    ///         .enumerable(true)
+    ///         .configurable(true)
+    /// );
     /// ```
     #[inline]
-    pub fn register_global_property<K, V>(&mut self, key: K, value: V, attribute: Attribute)
+    pub fn register_global_property<K, P>(&mut self, key: K, property: P)
```

Comment:
Here too we want to register a complete property (data descriptor), not a property descriptor.

Reply:
Fixed